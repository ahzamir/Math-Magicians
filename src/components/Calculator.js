import React from 'react';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  onClickHandler = (e) => {
    const buttonName = e.target.innerText;
    this.setState((prevValue) => calculate({
      ...prevValue,
    }, buttonName));
  };

  result = () => {
    const { total, operation, next } = this.state;
    return (
      <p>
        {total}
        {' '}
        {operation}
        {' '}
        {next}
      </p>
    );
  }

  render() {
    return (
      <div className="calculator-div">
        <div className="displayNum" id="result">{this.result()}</div>
        <div className="calculatorButtons">
          <button className="color-first" type="submit" onClick={this.onClickHandler}>AC</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>+/-</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>%</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>รท</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>7</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>8</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>9</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>x</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>4</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>5</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>6</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>-</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>1</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>2</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>3</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>+</button>
          <button className="color-second zero" type="submit" onClick={this.onClickHandler}>0</button>
          <button className="color-second" type="submit" onClick={this.onClickHandler}>.</button>
          <button className="color-first" type="submit" onClick={this.onClickHandler}>=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
