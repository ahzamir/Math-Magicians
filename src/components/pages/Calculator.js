import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const onClickHandler = (e) => {
    const buttonName = e.target.innerText;
    setState(calculate({
      ...state,
    }, buttonName));
  };

  const { total, operation, next } = state;

  const result = (
    <p>
      {total}
      {' '}
      {operation}
      {' '}
      {next}
    </p>
  );
  return (
    <div className="calculator-div">
      <div className="displayNum" id="result">{result}</div>
      <div className="calculatorButtons">
        <button className="color-first" type="submit" onClick={onClickHandler}>AC</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>+/-</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>%</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>รท</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>7</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>8</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>9</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>x</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>4</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>5</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>6</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>-</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>1</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>2</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>3</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>+</button>
        <button className="color-second zero" type="submit" onClick={onClickHandler}>0</button>
        <button className="color-second" type="submit" onClick={onClickHandler}>.</button>
        <button className="color-first" type="submit" onClick={onClickHandler}>=</button>
      </div>
    </div>
  );
};

export default Calculator;
