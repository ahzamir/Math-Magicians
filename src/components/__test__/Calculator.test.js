import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Calculator from '../pages/Calculator';

describe('Calculator Page testing', () => {
  render(<Calculator />);
  test('Testing the dom for success rendering elements on Calculator page', () => {
    expect(screen.queryByTestId('calculatorPageContent')).toBeInTheDocument();
  });

  test('Testing the dom for non existing elemente on the page after rendering', () => {
    expect(screen.queryByTestId('nothing')).not.toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Calculator page="just for test">Calculator</Calculator>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
