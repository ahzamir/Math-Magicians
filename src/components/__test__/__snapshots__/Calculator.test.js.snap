// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Page testing renders correctly 1`] = `
<div
  className="border calculatorPage"
  data-testid="calculatorPageContent"
>
  <h2
    className="headerText"
  >
    Lets do some math!
  </h2>
  <div
    className="calculator-div border"
  >
    <div
      className="displayNum"
      id="result"
    >
      <p>
         
         
      </p>
    </div>
    <div
      className="calculatorButtons"
    >
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        AC
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        +/-
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        %
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        รท
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        7
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        8
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        9
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        x
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        4
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        5
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        6
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        -
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        1
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        2
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        3
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
      <button
        className="color-second zero"
        onClick={[Function]}
        type="submit"
      >
        0
      </button>
      <button
        className="color-second"
        onClick={[Function]}
        type="submit"
      >
        .
      </button>
      <button
        className="color-first"
        onClick={[Function]}
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
